src/Provider/Provider.Web/Controllers/CloneVacancyController.cs

```diff
@@ -0,0 +1,89 @@
+using System.Threading.Tasks;
+using Esfa.Recruit.Provider.Web.Configuration.Routing;
+using Esfa.Recruit.Provider.Web.RouteModel;
+using Microsoft.AspNetCore.Mvc;
+using Esfa.Recruit.Provider.Web.Configuration;
+using Esfa.Recruit.Provider.Web.Orchestrators;
+using Esfa.Recruit.Provider.Web.ViewModels.CloneVacancy;
+using Esfa.Recruit.Provider.Web.Extensions;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Shared.Web.Extensions;
+using Esfa.Recruit.Shared.Web.ViewModels;
+
+namespace Esfa.Recruit.Provider.Web.Controllers
+{
+    [Route(RoutePaths.AccountVacancyRoutePath)]
+    public class CloneVacancyController : Controller
+    {
+        private readonly CloneVacancyOrchestrator _orchestrator;
+        public CloneVacancyController(CloneVacancyOrchestrator orchestrator)
+        {
+            _orchestrator = orchestrator;
+        }
+
+        [HttpGet("clone", Name = RouteNames.CloneVacancy_Get)]
+        public async Task<IActionResult> Clone(VacancyRouteModel vrm)
+        {
+            var vacancy = await _orchestrator.GetCloneableAuthorisedVacancyAsync(vrm);
+            
+            return _orchestrator.IsNewDatesRequired(vacancy) 
+                ? RedirectToRoute(RouteNames.CloneVacancyWithNewDates_Get) 
+                : RedirectToRoute(RouteNames.CloneVacancyDatesQuestion_Get);
+        }
+
+        [HttpGet("clone-dates-question", Name = RouteNames.CloneVacancyDatesQuestion_Get)]
+        public async Task<IActionResult> CloneVacancyDatesQuestion(VacancyRouteModel vrm)
+        {
+            var vm = await _orchestrator.GetCloneVacancyDatesQuestionViewModelAsync(vrm);
+            return View(vm);
+        }
+
+        [HttpPost("clone-dates-question", Name = RouteNames.CloneVacancyDatesQuestion_Post)]
+        public async Task<IActionResult> CloneVacancyDatesQuestion(CloneVacancyDatesQuestionEditModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                var vm = await _orchestrator.GetCloneVacancyDatesQuestionViewModelAsync(model);
+                return View(vm);
+            }
+
+            if (model.HasConfirmedClone == true)
```

Comment:
is the `==true` needed?

Reply:
yes since it is nullable... and it reads better